blog/2020-11-05-HowMuchFasterIsJava15.adoc

```diff
@@ -0,0 +1,105 @@
+= How much faster is Java 15?
+:page-interpolate: true
+:awestruct-author: mtomco
+:awestruct-layout: blogPostBase
+:awestruct-tags: [production, benchmark]
+
+Java 15 was released on the 15th of September 2020 and has promised to bring with it a few performance tweaks in its G1GC
+and ParallelGC garbage collectors. What does it mean for
+https://www.optaplanner.org/[OptaPlanner]? Have the improvements made in the G1GC outperform the reigning champion since Java 8 or does the ParallelGC still
+reign supreme on most use-cases? Is ParallelGC even faster than it was the previous time? Let's put it to the test!
+
+This article is a follow up on our
+https://www.optaplanner.org/blog/2019/01/17/HowMuchFasterIsJava11.html[ Java 11]
+and our
+https://www.optaplanner.org/blog/2014/03/20/HowMuchFasterIsJava8.html[Java 8] performance comparison benchmarks.
+
+== Benchmark methodology
+
+To run the benchmark we used:
+
+* A stable machine without any other computational demanding processes running and with
+`Intel® Xeon® Silver 4116 @ 2.1 GHz (12 cores total / 24 threads)` and `128 GiB` RAM memory, running `RHEL 8 x86_64`.
+
+* Both G1 and Parallel GC for both Java versions to compare the impact of garbage collection.
+Java executed with the parameters `-Xmx1536M -server -XX:+UseG1GC` and `-Xmx1536M -server -XX:+UseParallelGC` respectively.
+
+* Both OpenJDK 11 version "11.0.6"
+OpenJDK Runtime Environment 18.9 (build 11.0.6+10-LTS)
+OpenJDK 64-Bit Server VM 18.9 (build 11.0.6+10-LTS, mixed mode)
+
+* Both OpenJDK 15 version "15.0.1"
+OpenJDK Runtime Environment 18.9 (build 15.0.1+9-LTS)
+OpenJDK 64-Bit Server VM 18.9 (build 15.0.1+9-LTS, mixed mode)
+
+* OptaPlanner `7.44.0.Final`
+
+* Solving a planning problem involves *no IO* (except a few milliseconds during startup to load the input). *A single
+CPU is completely saturated.* It constantly creates many short-lived objects, and the GC collects them afterwards.
+
+* Each run solves 11 planning problems with OptaPlanner. Each planning problem runs for 5 minutes and starts with a
+30 second JVM warm up which is discarded.
+
+* The benchmarks measure the number of scores calculated per millisecond. Higher is better. Calculating
+a score for a proposed planning solution is non-trivial: it involves many calculations, including checking for
+conflicts between every entity and every other entity.
+
+== Executive summary
+
+With Java 15, the average improvement is *10.5%* for G1 and *6.2%* for Parallel GC. The difference between the two
+garbage collectors running on JDK 15 is *10.7%* leaning in favor of Parallel GC.
+For more information about difference between various GC algorithms, please see
+https://dzone.com/articles/choosing-the-best-garbage-collection-algorithm-for[this article].
```

Comment:
Generic link text. This is bad for accessibility and SEO. For example, if you replace this with "please see the following article that compares [Java garbage collectors performance](https://dzone.com/articles/choosing-the-best-garbage-collection-algorithm-for)", the link text will help search engines recognize this blog post is about Java GC performance.

- https://web.dev/link-text/
- http://jkorpela.fi/www/click.html

Reply:
I generally avoid external links except to Wikipedia and other, reliable open source projects, to avoid 404's in a few years when that content disappears (that then hurt our SEO ranking). DZone is pretty reliable though on not breaking their url's.